File: problems/2660_determine_the_winner_of_a_bowling_game/Solution.java
class Solution {
    public int isWinner(int[] p1, int[] p2) {
        int s1 = score(p1), s2 = score(p2);
        return s1 > s2 ? 1 : s2 > s1 ? 2 : 0;
    }
    int score(int[] p) {
        int s = 0;
        for (int i = 0; i < p.length; i++) {
            int m = (i>=1&&p[i-1]==10)||(i>=2&&p[i-2]==10) ? 2 : 1;
            s += m * p[i];
        }
        return s;
    }
}

File: problems/2660_determine_the_winner_of_a_bowling_game/solution.py
from typing import List


class Solution:
    def isWinner(self, player1: List[int], player2: List[int]) -> int:
        def score(p):
            s = 0
            for i in range(len(p)):
                mult = 2 if (i >= 1 and p[i-1] == 10) or (i >= 2 and p[i-2] == 10) else 1
                s += mult * p[i]
            return s
        s1, s2 = score(player1), score(player2)
        if s1 > s2: return 1
        if s2 > s1: return 2
        return 0
